src/main/java/org/spongepowered/common/util/ReflectionUtil.java

```diff
@@ -77,6 +84,32 @@
         Entity.class
     };
 
+    private static final String REFLECTION_PROPERTY_FILE = "reflection_mappings.json";
+    private static Map<String, String> mappingCache;
+
+    private static String getMethodNameForEnvironment(final String methodName) {
+        if (Launch.instance().developerEnvironment()) {
+            return methodName;
+        }
+
+        if (mappingCache == null) {
+            try (InputStreamReader reader = new InputStreamReader(Objects.requireNonNull(ReflectionUtil.class.getClassLoader().getResourceAsStream(REFLECTION_PROPERTY_FILE)))) {
+                mappingCache = new Gson().fromJson(reader, new TypeToken<Map<String, String>>(){}.getType());
+            } catch (IOException | NullPointerException e) {
+                mappingCache = Collections.emptyMap();
+            }
+        }
+
+        final String methodNameForEnvironment = mappingCache.get(methodName);
+
+        if (methodNameForEnvironment == null) {
+            SpongeCommon.logger().fatal(ReflectionUtil.STUPID_REFLECTION, "Could not find reflection mapping for method {} in reflection property file {}", methodName, REFLECTION_PROPERTY_FILE);
```

Comment:
I know this isn't your fault here - it's not really targeted at you Logic, but I'd rather not have the logger spit out "REFLECTION_BULLSHIT" every time this fails...

That said, do we _really_ need to add an entry into the Json file for every field that we use with reflection? The reality is that in SV our dev and prod environments are mapped the same, and this will be the case in Forge for 1.17.x and beyond. I don't really see why we should be emitting a _fatal_ logger message when it doesn't matter in that environment.

Is there a way to only generate this logger message if the environment is obfuscated?

Also, `REFLECTION_PROPERTY_FILE` static access needs to be qualified.

Reply:
To hard code, some platforms do not sound right to me, since this is common code. I guess it either introduces a new property `isObfuscated` in somewhere so I can use it to detect if the environment is obf or not. Or just put the json file even if it is not in obf environment.

Reply:
@dualspiral 

This will not be the case for Forge 1.17.x as production will be SRG mapped.

Anyways I agree with dual's comments, he brings up good points.

1. Per the REFLECTION_BULLSHIT comment...While funny, it isn't professional. Looking at you @gabizou 
2. We can add a `Launch` `requiresReflectionChecks` or something. `obfuscated` is not the right word here, some environments are just remapped vs. others.

Reply:
> This will not be the case for Forge 1.17.x as production will be SRG mapped with Official class names.

Ah, my misunderstanding. Good to know.

Reply:
> We can add a Launch requiresReflectionChecks

Should this be covered in this PR, or you will handle it later? Also, I guess `requiresReflectionRemap` is a better name?

Reply:
Chuck it into PR. Call it `Launch.usesThirdPartyMappings()`. Document method to state that this allows a platform to indicate that the codebase will be in mappings that are not supplied by Mojang. Something to that effect. Utilize that check and remove the mappings file from `vanilla`